src/screens/Home/surveyList.js

```diff
@@ -1,53 +1,102 @@
-import React, { useEffect, useRef } from 'react';
+import React, { useContext, useEffect, useRef, useState } from 'react';
 
 import Slider from 'react-slick';
 
-import testSurvey from 'images/auth_background.png';
+import { AuthContext } from 'contexts/auth';
+import { BackgroundContext } from 'contexts/background';
 import slickSettings from 'screens/Home/slickSettings';
+import Api from 'utils/Api';
+
+const SurveyList = ({ surveyResponse }) => {
+  const surveyData = surveyResponse.data;
+  const surveyMeta = surveyResponse.meta;
+  const total_pages = surveyMeta.pages;
 
-const SurveyList = () => {
   const sliderRef = useRef(null);
 
+  const [currentPage, setCurrentPage] = useState(surveyMeta.page);
+  const [list, setList] = useState(surveyData);
+  const [currentBackground, setCurrentBackground] = useState(
+    list[0].attributes.cover_image_url
+  );
+
+  const { dispatch } = useContext(BackgroundContext);
+  const authContext = useContext(AuthContext);
+
+  const sliderEvents = {
+    beforeChange: (current, next) => beforeSlideChange(next),
+    afterChange: (current) => afterSlideChange(current),
+  };
+  const slickConfig = { ...slickSettings, ...sliderEvents };
+
   useEffect(() => {
     window.addEventListener('wheel', (e) => {
       slideList(e.wheelDelta);
-    })
-  }, []);
+    });
+
+    dispatch({
+      type: 'SURVEY',
+      payload: currentBackground,
+    });
+  }, [currentBackground, dispatch]);
+
+  const beforeSlideChange = (next) => {
+    setCurrentBackground(list[next].attributes.cover_image_url);
+  };
+
+  const afterSlideChange = () => {
+    if (currentPage < total_pages) {
+      getList();
+    }
+  };
+
+  const getList = async () => {
+    const next_page = currentPage + 1;
+    try {
+      await Api.get(`api/v1/surveys?page[number]=${next_page}`, {
```

Comment:
Can we create a separate adapter for API calls and put it in `/adapters` just like the project structure mentioned in [campass](https://nimblehq.co/compass/development/code-conventions/react/#project-structure)
Maybe we can name it `SurveyAdapter` and provide an `index` function there to call the API while passing the page number param. 🤔

Reply:
Added in [a8f223d](https://github.com/rafayet-monon/surveyor/pull/17/commits/a8f223de1cc27e5b77de98df3267ec3217671b4b)